IPython/core/magics/namespace.py

```diff
@@ -460,8 +459,8 @@ def type_name(v):
                 try:
                     vstr = str(var)
                 except UnicodeEncodeError:
-                    vstr = unicode_type(var).encode(DEFAULT_ENCODING,
-                                               'backslashreplace')
+                    vstr = var.encode(DEFAULT_ENCODING,
+                                      'backslashreplace')
```

Comment:
I'm pretty sure this except block can't happen any more under Python 3, but leave it for now; I'll double check that and work out if anything else is needed instead.